src/graph/image.rs

```diff
@@ -134,11 +135,25 @@ pub enum CellWidthType {
 }
 
 impl ImageParams {
-    pub fn new(graph_color_set: &GraphColorSet, cell_width_type: CellWidthType) -> Self {
+    pub fn new(
+        graph_color_set: &GraphColorSet,
+        cell_width_type: CellWidthType,
+        detect_cell_size: bool,
+    ) -> Self {
+        let default_cell_size = (25, 50);
+        let (cell_width, cell_height) = match detect_cell_size {
+            true => get_cell_dimensions().unwrap_or(default_cell_size),
+            false => default_cell_size,
+        };
         let (width, height, line_width, circle_inner_radius, circle_outer_radius) =
             match cell_width_type {
-                CellWidthType::Double => (50, 50, 5, 10, 13),
-                CellWidthType::Single => (25, 50, 3, 7, 10),
+                CellWidthType::Double => {
+                    let inner = cell_width * 2 / 5;
+                    (cell_width * 2, cell_height, 5, inner, inner + 3)
+                }
+                CellWidthType::Single => {
+                    (cell_width, cell_height, 1, cell_width / 3, cell_width / 2)
+                }
```

Comment:
Does dynamically calculating cell sizes have anything to do with this fix in the first place?